### PR TITLE
[CI:DOCS] Update basic_networking.md to include a 2nd Example for Rootless Networking

### PR DESCRIPTION
**Pull request by reo-1963**
If creating a network is going to have no "network_dns_servers" by default, you guys should describe how to make this work in detail so noobs like me don't start thinking we ought to go back to Docker.  By all means have a real Tech Writer and a real Network Engineer review my proposed change. #18530

Richard Oberuc

#### Does this PR introduce a user-facing change?

None

### CODE REVIEW

**Comment by timway** [comment]
I'm not sure I understand this example. Looking at Fedora running rootless containers this is what I built to model what you are doing:

```bash
#! /usr/bin/bash

LOOKUP="nslookup -type=aaaa quad9.net"

CREATED_NETWORK=0
if ! podman network ls --filter name=testing --noheading | grep testing > /dev/null
then
    CREATED_NETWORK=1
    podman network create testing > /dev/null
fi

for TYPE in default podman testing
do
    NETWORK=""
    if [ "$TYPE" != "default" ]
    then
        NETWORK="--network $TYPE"
    fi

    echo "-*-*-*- Testing: $TYPE -*-*-*-"
    ( set -x ; podman run --detach --name dns-rootless-net-$TYPE $NETWORK --quiet --rm registry.access.redhat.com/ubi9/ubi-init > /dev/null)
    podman exec dns-rootless-net-$TYPE dnf --assumeyes install bind-utils ncurses > /dev/null
    echo $LOOKUP
    ( set -x ; podman exec --tty dns-rootless-net-$TYPE $LOOKUP | \
        python3 -c "import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\n')")
    ( set -x ; podman inspect dns-rootless-net-$TYPE | jq .[].HostConfig.NetworkMode)
    podman stop dns-rootless-net-$TYPE > /dev/null

    if [ "$TYPE" != "testing" ]
    then
        echo
        echo
    fi
done

if [ $CREATED_NETWORK != 0 ]
then
    podman network rm testing > /dev/null
fi
```

It results in:

```bash
$ bash blah.sh 
-*-*-*- Testing: default -*-*-*-
+ podman run --detach --name dns-rootless-net-default --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-default nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.0.2.3
Address:        10.0.2.3#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-default
+ jq '.[].HostConfig.NetworkMode'
"slirp4netns"


-*-*-*- Testing: podman -*-*-*-
+ podman run --detach --name dns-rootless-net-podman --network podman --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-podman nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.211.11.254
Address:        10.211.11.254#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-podman
+ jq '.[].HostConfig.NetworkMode'
"bridge"


-*-*-*- Testing: testing -*-*-*-
+ podman run --detach --name dns-rootless-net-testing --network testing --quiet --rm registry.access.redhat.com/ubi9/ubi-init
nslookup -type=aaaa quad9.net
+ podman exec --tty dns-rootless-net-testing nslookup -type=aaaa quad9.net
+ python3 -c 'import sys; sys.stdout.write(sys.stdin.read().rstrip() + '\''\n'\'')'
Server:         10.89.1.1
Address:        10.89.1.1#53

Non-authoritative answer:
Name:   quad9.net
Address: 2620:0:871:9000::77
+ podman inspect dns-rootless-net-testing
+ jq '.[].HostConfig.NetworkMode'
"bridge"
```

TLDR; I don't observe the broken DNS stack in any case. Is this just an issue in Podman machines or possibly in WSL2? Limited to VPN or similar conditions?

Also this PR is against your fork not the main repo like I think you intended to do.

**Comment by klesh** [comment]
Seems like this PR is targeting your own `main` branch. Shouldn't it be the [upstream](https://github.com/containers/podman) instead?
image

**Comment by reo-1963** [comment]
I'm not sure about where the "pull" or "merge" goes. My "Git" skills are not very good.

The problem with setting the DNS on a "network create" is very specific to just happening on WSL2 on Windows. #18530.
But if WSL2 is how Win 10 & Win 11 are going to install Podman (and will be an issue going forward), and #18530 has been an "unresolved" issue for the past 3 months, and it was a problem for me as a "noob user", it's going to be a problem for a lot of other new Users who may decide it's a deal breaker and prematurely walk away from Podman.

Maybe this should go into a "TroubleShooting" for "WSL2 and Windows and Networking and Rootless"? 
But not finding one of those, I am suggesting a 2nd example under "Networking and Rootless".

**Comment by klesh** [comment]
@reo-1963 It's fine, you just need to edit the PR and change the `base` to `containers/podman` so your proposal can be discussed and processed by the Podman maintainers.
